spatialmath/geometry.go

```diff
@@ -24,6 +24,7 @@ type Geometry interface {
 	Label() string   // Label is the name of the geometry
 	String() string  // String is a string representation of the geometry data structure
 	ToPoints(float64) []r3.Vector
+	Dimensions() []float64
```

Comment:
Agree with Peter's long comment about the bufferedWorldState, which will mean this is not necessary.  In general the "dimensions" of a Geometry are hard to generalize about since different representations will have outputs that mean different things